doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+`apt` for the package manager for your flavour of Linux.)
+
+The following will install the mandatory tools for building Taisei.
+
+.. code:: sh
+
+   apt update
+   apt install meson cmake build-essential
+
+
+The dependencies below are technically optional, as mentioned above in
+``Built-In Vs. System Dependencies``.
+
+Keep in mind that certain packages may have different names on different
+systems. Ensure you are installing the ``dev`` versions of those packages.
+
+.. code:: sh
+
+   apt install libsdl2-dev libsdl2-mixer-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev
+
+If your distribution of Linux uses Wayland as its default window server, ensure
+that Wayland deps are installed:
+
+.. code:: sh
+
+   apt install libwayland-dev
+
+macOS
+'''''
+
+On macOS, you must install the Xcode Command Line Tools to build Taisei for
+the platform, as it contains headers and tools that aren't included in the FOSS
+versions found in ``brew``.
+
+.. code:: sh
+
+   xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+.. code:: sh
+
+   brew install meson cmake pkg-config docutils imagemagick pygments
+
+The following dependencies are technically optional, and can be pulled in at
+build-time, but they will reduce compile times during development, so it's
+recommended to install them.
+
+.. code:: sh
+
+   brew install freetype2 libzip opusfile libvorbis webp sdl2
+
+Optionally, if you're on macOS and compiling for macOS, you can to install
+`create-dmg <https://github.com/create-dmg/create-dmg>`__, which will allow
+you to have nicer-looking macOS ``.dmg`` files for distribution:
+
+.. code:: sh
+
+   brew install create-dmg
+
+As of 2021-08-05, you should **not** install the following packages via
+Homebrew, as the versions available do not compile against Taisei correctly.
+If you're having mysterious errors, ensure that they're not installed.
+
+-  ``spirv-tools``
+-  ``spirv-cross``
+-  ``sdl2_mixer``
+
+.. code:: sh
+
+   brew remove spirv-tools spirv-cross sdl2_mixer
+
+In addition, if you're trying to compile on an older version of macOS
+(i.e: <10.12), SDL2 may not compile correctly on Homebrew (as of 2019-02-19).
+Let ``meson`` pull in the corrected version for you via subprojects.
+
+**NOTE:** While Homebrew's optional dependencies greatly improve compile times,
+if you can't remove packages that give you errors from your system for whatever
+reason, you can force ``meson`` to use its built-in subprojects by using
+``--wrap-mode`` (more on that later).
+
+Windows
+'''''''
+
+Taisei uses `mstorsjo/llvm-mingw <https://github.com/mstorsjo/llvm-mingw>`__ to
+achieve cross-compiling on Windows. Cross-compiling for Windows ends up being
+easier to maintain and more consistent than attempting to use Microsoft's native
+toolchain.
```

Comment:
We use that toolchain for official releases, but any mingw64-based toolchain should work fine. Some Linux distros package GCC-based mingw64 cross toolchains and many precompiled windows libraries.

Native compilation on windows is also possible with a ming64 toolchain (including that LLVM-based one that we use). But it's more complicated that it needs to be, mostly because of the symlinks in the repo. It's not "officially supported" (we do not endorse self harm) so I wouldn't expand on it in all the gory details here, but probably worth mentioning that it's possible to do if you hate yourself enough.

Also microsoft's toolchain isn't just inconvenient, their compiler just can not build Taisei. Maybe `clang-cl` can, but I never got that to work. Even I don't hate myself *that* much.